### ISSUE TITLE
Scripts use wrong python3

### ISSUE THREAD

**Issue by apt1002** [rrthomas/linton#6]
The Python scripts under `doc`, i.e. `*.in.py`, have a hash bang line `#!/usr/bin/env python3`, which may not be the same Python that Linton uses.

**Comment by rrthomas** [comment]
The obviously right thing to do would be to write the hash-bang line of the scripts at install time to point to the same Python interpreter as Linton is installed to use. (This is correct because these scripts should only ever be run in the scope of a Linton call, so that Python interpreter should always be present when they are run.)

However, the only way I know how to do this is to make these scripts entry-points for Linton itself, which seems wrong. (In this case, pip will generate wrapper scripts, much as it does for Linton *ipse*, using the correct Python interpreter).  In particular, it is not a model that users writing their own scripts can copy.

I could simply check the version of Python at run-time and error out if it's wrong. But that will probably confuse users who don't understand why having installed Linton without problem it won't run correctly. (I could give a suitably detailed error to help, of course.)

Coming at it from the other end, what would I expect users to do when writing their own Python scripts? The typical case will be that they will run Linton with their default Python. In this case, having the current hash-bang line will work fine. Of course if they explicitly want to use a different Python (e.g. Python 2!) they can use a different hash-bang line, and that's on them, just as if they wrote a script in Perl, shell etc.

I could require that `/usr/bin/env python3` be a suitable version for the scripts, and issue an install-time error if that is not so (effectively, declaring a separate dependency of the scripts on the Python version used, even though in practice I will keep the scripts compatible with the versions of Python that Linton itself supports).

One other thought is that if I implement the trick I mentioned before, of running Python scripts in-process (unlike all other scripts), then this will not be a problem.

One thought: Linton could add a temporary entry to `PATH` so that `/usr/bin/env python3` is guaranteed to be Linton's Python. Or, a command `linton python …` could run a Python script with Linton's Python.

**Comment by rrthomas** [comment]
> A command `linton python …` could run a Python script with Linton's Python.

This could serve a double purpose: scripts could use it to get the same Python that Linton is using (this is really only of interest to scripts that come with Linton); and, scripts could use it to mark themselves as amenable to optimisation by being run in-process; it seems sensible to have an opt-in for this optimisation rather than simply trying to run all Python scripts in-process.

**Comment by apt1002** [comment]
I don't think you should make any design decisions that expose the true but circumstantial fact that Linton is implemented in Python.

**Comment by rrthomas** [comment]
The scripts can declare a dependency, as this is now supported. See how we do this in chambercourt with the levels conversion scripts.
